packages/valory/skills/proposal_collector_abci/rounds.py

```diff
@@ -267,7 +270,11 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
                 SynchronizedData, self.synchronized_data
             ).snapshot_proposals
             snapshot_proposals.extend(payload["snapshot_proposals"])
-            print(f"Adding {len(payload['snapshot_proposals'])}")
+            finished = (
+                payload["finished"]
+                or len(snapshot_proposals) >= SNAPSHOT_PROPOSAL_TOTAL_LIMIT
+            )
+
```

Comment:
Even if there are still proposals to retrieve, the first 200 are more than enough for one period.